pkg/installation_test.go

```diff
@@ -57,28 +37,6 @@ func TestCreateInstallationResources(t *testing.T) {
 			HasNoOwnerRef()
 	})
 
-	t.Run("when subscription is set as owner reference then it should be removed", func(t *testing.T) {
-		// given
-		tektonInstallation := tektoninstallation.NewInstallation()
-		err := controllerutil.SetControllerReference(subscription, tektonInstallation, s)
-		require.NoError(t, err)
-		cheInstallation := cheinstallation.NewInstallation()
-		err = controllerutil.SetControllerReference(subscription, cheInstallation, s)
-		require.NoError(t, err)
-
-		client := test.NewFakeClient(t, subscription, tektonInstallation, cheInstallation)
-
-		// when
-		err = pkg.CreateInstallationResources(client, s, logf.Log)
-
-		// then
-		require.NoError(t, err)
-		assert.AssertThatTektonInstallation(t, "", tektoninstallation.InstallationName, client).
-			HasNoOwnerRef()
-		assert.AssertThatCheInstallation(t, "", cheinstallation.InstallationName, client).
-			HasNoOwnerRef()
-	})
-
```

Comment:
In the new controller-runtime we can't now set the namespaced owner reference (Subscription) to a cluster scoped resource (CheInstallation):
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L67
So, the test was failing.
We could re-implement `controllerutil.SetControllerReference()` in our tests though but I think it would be overkill. The controllerutil is now taking care of making sure we don't set such broken references so no need to spend too much time on re-testing it on our side.

Reply:
ok, make sense. Thanks for explanation :-)

Reply:
it's good news that the controller-runtime can detect and prevent such thing now. We got trapped in the past and this will not happen again from now on ;)